docs/blog/posts/2020-03-06-wordEmbeddings.md

```diff
@@ -0,0 +1,113 @@
+---
+date:
+  created: 2020-03-04
+categories:
+    - Machine Learning
+    - GPU
+authors:
+    - nrapstin
+
+ 
+---
+
+# Word Embeddings
+
+## What are Word Embeddings?
+
+Word Embeddings are a method to translate a string of text into an N-dimensional vector of real numbers.  Many computational methods are not capable of accepting text as input.  This is one method of transforming text into a number space that can be used in various computational methods.
+
+<!-- more -->
+
+## An Example
+
+| Word  | "Royalty" Feature | "Masculinity" Feature |
+| ----- | ----------------- | --------------------- |
+| Queen | 0.958123          | 0.03591294            |
+| King  | 0.94981289        | 0.92959219            |
+| Man   | 0.051231          | 0.9592109321          |
+| Woman | 0.0912987         | 0.04912983189         |
+
+Let's say we were to create embeddings for the four words above, with two features representing Royalty and Masculinity.  The resulting embedding for the word "Queen" is <0.958123, 0.03591294> - a value close to 1 for Royalty, and a value close to 0 for Masculinity.  In this example, we could identify the concept of Masculinity by the vector <0,1> and Royalty by <1,0>.
+
+| Word  | Feature 1 | Feature 2 | Feature 3 | Feature 4 |
+| ----- | --------- | --------- | --------- | --------- |
+| Queen | 0.854712  | 0.32145   | 0.123512  | 0.023051  |
+| King  | 0.9521    | 0.23151   | 0.721385  | 0.950213  |
+| Man   | 0.12351   | 0.067231  | 0.85921   | 0.821231  |
+| Woman | 0.23131   | 0.123151  | 0.011249  | 0.23151   |
+
+However, in real world examples, it is rarely this simple.  In general, we do not know exactly what each of the N features mean.  It is very common to test the quality of embeddings using analogies.  For example, even though we may not be able to identify a feature for Royalty, we expect that subtracting the vectors "King - Man" and "Queen - Woman" should give us roughly the same vector representing Royalty.  Similarily for Masculinity, we can test the similarity of the vectors "Man - Woman" and "King - Queen".  
+
+## Common Embedding Models
+
+
+### Word2Vec
+Developed by Google, it uses skip-gram and continuous bag-of-words (CBOW) architectures to create word embeddings. [You can read more on Word2Vec's wikipedia page.](https://en.wikipedia.org/wiki/Word2vec) [The Gensim package in python has an implementation of these algorithms](https://radimrehurek.com/gensim/models/word2vec.html)
```

Comment:
make sure all links open up in new tabs. You are also missing a lot of end of sentence periods.